hack/build_artifacts.sh

```diff
@@ -0,0 +1,44 @@
+#! /bin/zsh
+
+setopt ERR_EXIT
+setopt PIPE_FAIL
+
+perhaps_module_dirs=(pkg/)
+
+for perhaps_module_dir in $perhaps_module_dirs; do
```

Comment:
@ransomw1c could you please introduce that one? what is this about? running tests on dependencies?
I am not even sure that `go mod download` picks up tests in dependency packages (since the introduction of the GO MOD PROXY feature, I'd say).

Reply:
> what is this about?

ensuring that the test binaries/executable build.

----

introduction for team members less familiar with golang is that (as you know) under the hood, `go test` builds an executable similar to `go build`, then runs that executable.  go, unlike python, is a typed and compiled language.  we get a reasonable amount of sanity checks by choosing appropriate types and making sure everything builds, runtime checks aside.

passing the `-c` flag to `go build` says, "build the test executable only, don't run it."  and `go test -c` can only build one executable at a time.  so this script is there to traverse all the test binaries and make sure everything is kosher thru compile time, stopping short of the runtime checks provided by `go test ./...` or somesuch.